python/metatensor-operations/metatensor/operations/_testing/_creation_operation.py

```diff
@@ -0,0 +1,93 @@
+from typing import Union
+
+import numpy as np
+
+from .. import _dispatch
+from .._classes import Labels, TensorBlock, TensorMap
+from .._dispatch import TorchTensor
+from ..block_from_array import block_from_array
+
+
+def cartesian_cubic(
+    cartesian_vector: Union[np.ndarray, TorchTensor], compute_grad: bool = False
+) -> TensorMap:
+    """
+    Creates a tensor map from a set of Cartesian vectors together with gradients if
+    :param compute_grad: is `True` according to the function:
+
+    .. math::
+
+        f(x, y, z) = x^3 + y^3 + z^3
+
+        \\nabla f = (3x^2, 3y^2, 3z^2)
+
+    :param cartesian_vector: Set of Cartesian vectors with shape (n_samples, 3)
+    :param compute_grad: Specifies if the returned tensor map should contain the
+        gradients
+    """
+
+    cartesian_vector_cubic = cartesian_vector**3
+    values = _dispatch.sum(cartesian_vector_cubic, axis=1).reshape(-1, 1)
+    if compute_grad:
+        values_grad = _dispatch.zeros_like(cartesian_vector, (len(values), 3, 1))
+        values_grad[:, 0] = 3 * cartesian_vector[:, 0:1] ** 2
+        values_grad[:, 1] = 3 * cartesian_vector[:, 1:2] ** 2
+        values_grad[:, 2] = 3 * cartesian_vector[:, 2:3] ** 2
+
+    block = block_from_array(values)
+    if compute_grad:
+        block.add_gradient(
+            parameter="positions",
+            gradient=TensorBlock(
+                values=values_grad,
+                samples=Labels.range("sample", len(values)),
+                components=[Labels.range("cartesian", 3)],
+                properties=block.properties,
+            ),
+        )
+    return TensorMap(Labels.range("_", 1), [block])
+
+
+def cartesian_linear(
```

Comment:
`cartesian_linear` is easier to debug errors, since it is just 3 different constant values. In sort this helps. While `cartesian_cubic` has a more complex pattern.

> It feels to me a more relevant distinction would be 1 gradient per sample or more than one gradient per sample, which could be done with

I can add this to the finite difference issue, but I would like to proceed with getting the fix for sort out, if you okay.

Reply:
Yeah, handling of different set of gradients samples is a longer term project!